app/services/concerns/xccdf/profile_rules.rb

```diff
@@ -24,20 +24,20 @@ def save_profile_rules
       def profile_rules
         @profile_rules ||= @op_profiles.flat_map do |op_profile|
           profile_id = profile_id_for(ref_id: op_profile.id)
-          rule_ids_for(ref_ids: op_profile.selected_rule_ids).map do |rule_id|
+          rule_ids_for(ref_ids: op_profile.selected_rule_ids).map do |entity_id|
             ::ProfileRule.find_or_initialize_by(
-              profile_id: profile_id, rule_id: rule_id
+              profile_id: profile_id, entity_id: entity_id
             )
           end
         end
       end
 
-      def links_to_remove(base)
+      def profile_rule_links_to_remove(base)
         grouped_rules = profile_rules.group_by(&:profile_id)
         grouped_rules.reduce(ProfileRule.none) do |query, (profile_id, prs)|
           query.or(
             base.where(profile_id: profile_id)
-                .where.not(rule_id: prs.map(&:rule_id))
+                .where.not(entity_id: prs.map(&:entity_id))
```

Comment:
Just a guess, but after polymorphism I think it will be looking something like:
```suggestion
                .where.not(entity: prs.map(&:entity))
```